docs/src/examples/advanced/asynchronous_jobs.md

```diff
@@ -0,0 +1,106 @@
+# Asynchronous jobs
+
+```@meta
+DocTestSetup = quote
+    ENV["ANYON_QUANTUM_USER"] = "test-user"
+    ENV["ANYON_QUANTUM_TOKEN"] = "not-a-real-token"
+    ENV["ANYON_QUANTUM_HOST"] = "yukon.anyonsys.com"
+end
+```
+
+This example is going to how to run jobs asynchronously using Julia tasks. Asynchronous jobs allow users to continue other work while waiting for the quantum resources.
+
+## Julia tasks
+
+Communication with a quantum computer happens over a network. Any requests sent to a quantum computer are sent over the network. After the request is sent, we can only continue the job execution when we get a response from the quantum computer.
+
+If we absolutely have to have the job's result, we have no choice but to wait. Often, other work can be done while waiting for the quantum job to complete. We want to be able to suspend the `run_job` so that we can continue with other work until we are ready to use the result. This type of control flow is called asynchronous control flow. We recommend you read [Julia's page on asynchronous programming](https://docs.julialang.org/en/v1/manual/asynchronous-programming/) if you are unfamiliar with it.
```

Comment:
By "suspend" I understand "pause". We want to be able to decouple the execution of `run_job` from the rest of our program?